build-tools/packages/build-cli/src/commands/generate/releaseNotes.ts

```diff
@@ -0,0 +1,202 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { writeFile } from "node:fs/promises";
+import path from "node:path";
+import { loadFluidBuildConfig } from "@fluidframework/build-tools";
+import { Flags } from "@oclif/core";
+import { StringBuilder } from "@rushstack/node-core-library";
+import { format as prettier } from "prettier";
+import { remark } from "remark";
+import remarkGfm from "remark-gfm";
+import remarkGithub, { defaultBuildUrl } from "remark-github";
+import admonitions from "remark-github-beta-blockquote-admonitions";
+import remarkToc from "remark-toc";
+
+import { releaseGroupFlag } from "../../flags.js";
+import {
+	BaseCommand,
+	DEFAULT_CHANGESET_PATH,
+	FluidCustomChangeSetMetadataDefaults,
+	UNKNOWN_SECTION,
+	difference,
+	groupBySection,
+	loadChangesets,
+} from "../../library/index.js";
+
+/**
+ * Generates release notes from individual changeset files.
+ */
+export default class GenerateReleaseNotesCommand extends BaseCommand<
+	typeof GenerateReleaseNotesCommand
+> {
+	static readonly summary = `Generates release notes from individual changeset files.`;
+
+	// Enables the global JSON flag in oclif.
+	static readonly enableJsonFlag = true;
+
+	static readonly flags = {
+		releaseGroup: releaseGroupFlag({
+			required: true,
+		}),
+		releaseType: Flags.custom<"major" | "minor">({
```

Comment:
This flag was inherited from the `upcoming` command, and was more needed when we had a next branch that contained both
minor and major changesets. We wanted to omit the minor changesets when runnning in next because they were going to be
gone by the time the major release was done.

The typical changesets workflow assumes you're working from one branch, and releasing major/minor/patches all from the
same branch. We don't use them that way. We treat all changes in a given release as the same release type - mostly
minor, now.

We can change how we use the changeset metadata, but that's a separate discussion and we can make follow up changes if needed.